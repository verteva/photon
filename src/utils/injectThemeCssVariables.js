export const injectThemeCssVariables = (flattenedTheme) => {
  const headTag = document.getElementsByTagName('head')[0];
  const styleTag =
    document.getElementById('theme-css-variables') ||
    document.createElement('style');

  styleTag.id = 'theme-css-variables';

  const reducedStyles = Object.entries(flattenedTheme).reduce(
    (acc, [key, value]) => {
      return acc + `${key}: ${value};`;
    },
    ''
  );

  styleTag.innerHTML = `
    :root {
      ${reducedStyles}
    }
  `;
  headTag.appendChild(styleTag);
};
