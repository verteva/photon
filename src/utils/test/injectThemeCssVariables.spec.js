import { injectThemeCssVariables } from '../injectThemeCssVariables';
import parsedTheme from './parsedTheme.json';

describe('injectThemeCssVariables.js', () => {
  it('injects the css into the dom', () => {
    injectThemeCssVariables(parsedTheme);
    expect(
      document.getElementById('theme-css-variables').textContent
    ).toMatchSnapshot();
  });
});
