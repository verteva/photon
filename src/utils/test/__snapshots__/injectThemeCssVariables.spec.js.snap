// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`injectThemeCssVariables.js injects the css into the dom 1`] = `
"
    :root {
      colors-brand-green: #219653;colors-brand-green-dark: #1d7a44;colors-brand-green-light: #2ebd60;colors-brand-green-lighter: #e6f4ec;
    }
  "
`;
